From 3e8a10addc1731eda2c4851e76ae9b025e4d5b4c Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Wed, 6 Sep 2017 15:46:00 +0800
Subject: [PATCH 3311/5242] MLK-17491-9 pinctrl: imx: cleanup
 gpio_set_direction

commit  fefd3119a6be7ae5ac87a52b0074d6639844211b from
https://source.codeaurora.org/external/imx/linux-imx.git

First of all, the design of using CONFIG_IBE_OBE is wrong as both VF and
IMX has IBE and OBE while current code defined it in common code but for
only IMX which causes a bit confusing.

Second, remove the following invalid comments as we will clear IBE.
"IBE always enabled allows us to read the value on the wire"

Last, replace the complicated "if else" statement with a much simpler one.

Cc: Bai Ping <ping.bai@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Fugang Duan <fugang.duan@nxp.com>
Fixes: 07787c40ff3b ("MLK-13485-3 pinctrl: imx: modify the imx pinctrl to support imx7ulp gpio")
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pinctrl/freescale/pinctrl-imx.h     |    4 ++--
 drivers/pinctrl/freescale/pinctrl-imx7ulp.c |    4 ++++
 drivers/pinctrl/freescale/pinctrl-memmap.c  |    3 ---
 drivers/pinctrl/freescale/pinctrl-vf610.c   |    2 ++
 4 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/pinctrl/freescale/pinctrl-imx.h b/drivers/pinctrl/freescale/pinctrl-imx.h
index 1c50b6e..03f1507 100644
--- a/drivers/pinctrl/freescale/pinctrl-imx.h
+++ b/drivers/pinctrl/freescale/pinctrl-imx.h
@@ -75,6 +75,8 @@ struct imx_pinctrl_soc_info {
 	/* MUX_MODE shift and mask in case SHARE_MUX_CONF_REG */
 	unsigned int mux_mask;
 	u8 mux_shift;
+	u32 ibe_bit;
+	u32 obe_bit;
 
 	/* generic pinconf */
 	bool generic_pinconf;
@@ -114,8 +116,6 @@ struct imx_pinctrl {
 
 #define SHARE_MUX_CONF_REG	0x1
 #define ZERO_OFFSET_VALID	0x2
-#define CONFIG_IBE_OBE		0x4
-
 #define IMX8_ENABLE_MUX_CONFIG	(1 << 29)
 #define IMX8_ENABLE_PAD_CONFIG	(1 << 30)
 #define IMX8_USE_SCU		(1 << 31)
diff --git a/drivers/pinctrl/freescale/pinctrl-imx7ulp.c b/drivers/pinctrl/freescale/pinctrl-imx7ulp.c
index 320a457..2db5a48 100644
--- a/drivers/pinctrl/freescale/pinctrl-imx7ulp.c
+++ b/drivers/pinctrl/freescale/pinctrl-imx7ulp.c
@@ -437,6 +437,8 @@ static int imx7ulp_pmx_gpio_set_direction(struct pinctrl_dev *pctldev,
 	.decodes = imx7ulp_cfg_decodes,
 	.num_decodes = ARRAY_SIZE(imx7ulp_cfg_decodes),
 	.fixup = imx7ulp_cfg_params_fixup,
+	.ibe_bit = BIT(16),
+	.obe_bit = BIT(17),
 };
 
 static const struct imx_pinctrl_soc_info imx7ulp_pinctrl_info_1 = {
@@ -449,6 +451,8 @@ static int imx7ulp_pmx_gpio_set_direction(struct pinctrl_dev *pctldev,
 	.decodes = imx7ulp_cfg_decodes,
 	.num_decodes = ARRAY_SIZE(imx7ulp_cfg_decodes),
 	.fixup = imx7ulp_cfg_params_fixup,
+	.ibe_bit = BIT(16),
+	.obe_bit = BIT(17),
 };
 
 static struct of_device_id imx7ulp_pinctrl_of_match[] = {
diff --git a/drivers/pinctrl/freescale/pinctrl-memmap.c b/drivers/pinctrl/freescale/pinctrl-memmap.c
index fc93976..cfc8b8f 100644
--- a/drivers/pinctrl/freescale/pinctrl-memmap.c
+++ b/drivers/pinctrl/freescale/pinctrl-memmap.c
@@ -30,9 +30,6 @@
 
 #define IMX_PAD_SION 0x40000000		/* set SION */
 
-#define IOMUXC_IBE	(1 << 16)
-#define IOMUXC_OBE	(1 << 17)
-
 int imx_pmx_set_one_pin_mem(struct imx_pinctrl *ipctl, struct imx_pin *pin)
 {
 	const struct imx_pinctrl_soc_info *info = ipctl->info;
diff --git a/drivers/pinctrl/freescale/pinctrl-vf610.c b/drivers/pinctrl/freescale/pinctrl-vf610.c
index 37602b05..75d0e54 100644
--- a/drivers/pinctrl/freescale/pinctrl-vf610.c
+++ b/drivers/pinctrl/freescale/pinctrl-vf610.c
@@ -320,6 +320,8 @@ static int vf610_pmx_gpio_set_direction(struct pinctrl_dev *pctldev,
 	.gpio_set_direction = vf610_pmx_gpio_set_direction,
 	.mux_mask = 0x700000,
 	.mux_shift = 20,
+	.ibe_bit = BIT(0),
+	.obe_bit = BIT(1),
 };
 
 static const struct of_device_id vf610_pinctrl_of_match[] = {
-- 
1.7.9.5

